### DISCUSSION TITLE
MaxLFQ values for samples with only one valid peptide ratio

### DISCUSSION THREAD

**Discussion by Mengbo-Li** [tvpham/iq#22] [General]
Hi, 
I have tried running iq using both the pure R version and the github version, but have noticed that both give me non-NA values for samples which are supposed to be NA. For a sample that has only one observed precursor, according to the original publication, it would not have valid peptide ratios so that it would be set to 0, hence NA after log2-transformation (sample F in Figure 2c from Cox et al 2014). 
Is there something different from the original methods that you are doing here?
Thanks, 
Mengbo

**Comment by tvpham** [comment]
Hi Mengbo,

It has to do with the filtering of the data. If you want to filter out data point with only one observed precursor, you can do that on the final data matrix.

In MaxQuant (the software that implements MaxLFQ), I think they filter out "ratios" that have only 1 pair. If you really want to do that, at the moment, you need to modify the code (line 320 ` r_i_j <- median(-X[,i] + X[,j], na.rm = TRUE)`), so that `r_i_j` is NA if there is only 1 pair. Maybe there will be an option to do that in the future release. But I am not sure if that is really needed.

Thang

**Reply by Mengbo-Li** [reply]
Hi Thang, 
Thank you a lot for the reply. 
The reason why I am asking is that, I notice that such a modification will have a great impact on the number of NA values in the results. IQ-summarised data will give much less NA values than the pg_matrix.tsv output by DIA-NN. I understand that there is a discrepancy among several implementations, which will lead to different quantification values. However, I thought at least a sample that is NA by one implementation should be also NA by another? Otherwise this is not merely a difference in the numerical accuracy anymore.  
Mengbo
